diagrams/lib/stroke_join.dart

```diff
@@ -0,0 +1,250 @@
+  final double endAngle;
+  final StrokeJoin join;
+  final double strokeMiterLimit;
+
+  @override
+  StrokeJoinPainterState createState() => new StrokeJoinPainterState();
+}
+
+class StrokeJoinPainterState extends State<StrokeJoinPainterWidget> //
+    with
+        TickerProviderStateMixin<StrokeJoinPainterWidget> {
+  AnimationController controller;
+
+  @override
+  void didUpdateWidget(StrokeJoinPainterWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    controller.value = 0.0;
+    controller.forward();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    controller = new AnimationController(
+      duration: widget.duration,
+      vsync: this,
+      lowerBound: widget.startAngle,
+      upperBound: widget.endAngle,
+    )..addListener(() {
+        setState(() {});
+      });
+    controller.forward();
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final StrokeJoinDescription description = new StrokeJoinDescription(
+      angle: controller.value,
+      join: widget.join,
+      strokeMiterLimit: widget.strokeMiterLimit,
+    );
+
+    return new ConstrainedBox(
+      constraints: new BoxConstraints.tight(const Size(300.0, 300.0)),
+      child: new Container(
+        padding: const EdgeInsets.all(18.0),
+        color: Colors.white,
+        child: new CustomPaint(painter: description),
+      ),
+    );
+  }
+}
+
+class StrokeJoinDiagramStep extends DiagramStep {
+  StrokeJoinDiagramStep(DiagramController controller) : super(controller);
+
+  @override
+  final String category = 'dart-ui';
+
+  @override
+  Future<List<File>> generateDiagrams() async {
+    final List<StrokeJoinPainterWidget> strokes = <StrokeJoinPainterWidget>[
+      const StrokeJoinPainterWidget(
+        filename: 'miter_0_join',
+        join: StrokeJoin.miter,
+        strokeMiterLimit: 0.0,
+      ),
+      const StrokeJoinPainterWidget(
+        filename: 'miter_4_join',
+        join: StrokeJoin.miter,
+        strokeMiterLimit: 4.0,
+      ),
+      const StrokeJoinPainterWidget(
+        filename: 'miter_6_join',
+        join: StrokeJoin.miter,
+        strokeMiterLimit: 6.0,
+      ),
+      const StrokeJoinPainterWidget(
+        filename: 'round_join',
+        join: StrokeJoin.round,
+      ),
+      const StrokeJoinPainterWidget(
+        filename: 'bevel_join',
+        join: StrokeJoin.bevel,
+      ),
+    ];
+
+    final List<File> outputFiles = <File>[];
+    for (StrokeJoinPainterWidget stroke in strokes) {
+      print('Drawing stroke diagram for ${stroke.join} (${stroke.filename})');
+      controller.builder = (BuildContext context) => stroke;
+      controller.filenameGenerator = () => new File(stroke.filename);
+      outputFiles.add(
+        await controller.drawAnimatedDiagramToFiles(end: const Duration(seconds: 6), frameDuration: const Duration(milliseconds: 200), metadata: <String, dynamic>{
```

Comment:
can we make this 60Hz rather than 5Hz? (i found the real one this time!)

Reply:
well this one is going to be an MPEG, right? So that's generally rendered streaming.

Reply:
Actually, after switching it to 60hz lossless encoding, optimized for cell animations, the file size is about 100K, so I don't think we need to worry. :-)